### DISCUSSION TITLE
prevent selector from firing on a OR clause, fire on AND clause of 2 atoms

### DISCUSSION THREAD

**Discussion by MxD-js** [facebookexperimental/Recoil#1722] [Q&A]
Hey All, 

Please excuse, if this is not the place to ask questions. 

Consider this, There are 2 atoms in the selector, this selector calls a back end API, but It needs both atoms state before firing a async request to the back end, is there a solution to have a "AND" clause that will ensure both atoms have state before firing off a request? I know that I can add "if" check if the state is "valid" before submitting a request, but wondering if there was a built in solution.

```
export const getTrendsForGivenLocationAndDate = selector({
  key: "getTrendsForGivenLocationAndDate",
  get: async ({ get }) => {
    console.log("getTrendsForGivenLocationAndDate function called");
    const datePickerS = get(datePickerState);
    const locations = get(checksComponentSelectedLocations);

    const response = await requestTrends(
      datePickerS.startDateMillis,
      datePickerS.endDateMillis,
      locations
    );

    console.log(response);
  },
});
```

**Comment by drarmstr** [comment]
The selector will only continue execution if a dependency is no longer pending.  So, the `requestTrends()` will only execute once both `datePickerState` and `checksComponntSelectedLocations` are no longer pending.  If they have a value but it is invalid in some application-specific way then you would have to check that.
